src/xrdok.js

```diff
@@ -681,3 +681,130 @@ const XRiconchevron = 'xr-icon-chevron';
   });
 
 })();
+
+const XRplanegraph = 'xr-plane-graph';
+(function () {
+
+  AFRAME.registerComponent(XRplanegraph, {
+
+    schema: {
+      color: { type: 'color', default: '#CCF'},
+      colorMax: { type: 'color', default: '#FCC'},
+      colorMin: { type: 'color', default: '#CCF'},
+      descriptors: {type: 'array', default: [ '1960', '1970', '1980', '1990', '2000', '2010', '2020' ] },
+      primaryValues: { type: 'array', default: [ 6.8, 7, 7.5, 7, 8, 9, 9.5 ] },
+      primaryValuesBase: { type: 'number', default: 0},
+      secondaryValues: { type: 'array', default: [ 41, 40, 35, 36, 32, 28, 10 ] },
+      secondaryValuesBase: { type: 'number', default: 3.65 },
+      segmentLength: { type: 'number', default: 2 },
+      showAsLine: { type: 'boolean', default: false},
+      src: { type: 'string', default: './models/lufttemperatur_bbb_sortiert.csv'},
+    },
+
+    init: function() {
+      // data and element
+      var data = this.data;
+      var el = this.el;
+
+      let self = this;
+
+      // parse CSV, if available
+      Papa.parse(data.src, {  // eslint-disable-line no-undef
+        download: true,
+        complete: function(res) {
+          data.descriptors = [];
+          data.primaryValues = [];
+          data.secondaryValues = [];
+          for (var i = 0; i < res.data.length-1; i++) {
+            data.descriptors.push(res.data[i][0]);
+            data.primaryValues.push(parseFloat(res.data[i][1]/10));
+            data.secondaryValues.push(parseInt(res.data[i][2]));
+          }
+
+          self.geometry = new THREE.Geometry();
+          
+          if (data.showAsLine) {
+            self.createGeometryLine(el, self, data);
+          } else {
+            self.createGeometryPlane(el, self, data);
+          }
+          
+          // create material
+          self.material = new THREE.MeshStandardMaterial();
+          self.material.side = THREE.DoubleSide;
+          self.material.transparent = true;
+          self.material.opacity = 1;
+
+          // create mesh
+          self.mesh = new THREE.Mesh(self.geometry, self.material);
+
+          // set mesh on entity
+          el.setObject3D('xr-plane-mesh', self.mesh);
+        }
+      });
+
+    },
+
+    remove: function() {
+      Object.values(this.children).forEach(v => this.el.removeChild(v));
+    },
+
+    createGeometryPlane: function(el, self, data) {
+      var offset = data.primaryValues[0];
+      
+      // create geometry
+      for (var j = 0; j < Math.min(data.descriptors.length-1, data.primaryValues.length-1); j++) {
+        // get const
+        const prim = data.primaryValues[j]-offset;
+        const sec = data.secondaryValuesBase * data.secondaryValues[j]/100;
+        const label = document.createElement('a-text');
+        const line = document.createElement('a-entity');
+        const linePath = document.createElement('a-entity');
+
+        // create vertices
+        self.geometry.vertices.push(
```

Comment:
instead of modifying/assigning members to self, try creating local variables that you return at the end of the function. if it is multiple variable, return an object with all of them.
Then the function caller can decide what to do with those vars. In your case you would assign them to self.